dali/util/image.h

```diff
@@ -36,45 +36,88 @@ namespace dali {
 
 class ImgSetDescr {
```

Comment:
We probably could move non-template methods implementations to .cc file from the header file. Previously the functions were implemented there.

Reply:
Because 
```
ImgSetDescr DALITest::imageDecrs_[kLastImgType];
```
is used in 
```
template <typename ImgType>
class DALISingleOpTest : public DALITest {
 public:
  inline void SetUp() override {
    const char *subDir[] = {"", "/png"};
    DALITest::SetUp();
    c_ = (IsColor(ImageType()) ? 3 : 1);
    imageDecrs_[t_jpegImgType].clear();

    const auto flags = GetImageLoadingFlags();

    for (int j = t_undefinedImgType; ++j < kLastImgType;) {
      const uint32_t mask = 1 << j;
      if (flags & mask) {
        if (j != t_pngImgType)
          imageDecrs_[j].LoadImages(image_folder + subDir[j]);
        else
          imageDecrs_[j].LoadImages("", &images::png_test_images);

```
it looks like it is somehow "indirectly" templated. After moving `ImgSetDescr::LoadImages(...)` into `image.cc` we are getting bunch of "undefined reference" errors.

Before, only some “predecessors” `ImgSetDescr::LoadImages()`, which were written as global functions were in image.cc. 

I will leave it as it is.